### PR TITLE
test(iframes): add x-frame-options display test

### PR DESCRIPTION
**Pull request by rwoll**
Firefox is marked as failing since Playwright's build of Firefox happily
displays the iframe while the consumer build of Firefox refuses display
of the iframe.

### CODE REVIEW

**Comment by rwoll** [comment]
@dgozman Here's a new MR with the one test that was showing a unique issue. I've also removed the clicky-portions since that's a separate and unrelated concern.

File: test/frame.jest.js [hunk 1]
```diff
@@ -265,4 +265,20 @@ describe('Frame Management', function() {
...
+  it.fail(FFOX)('should refuse to display x-frame-options:deny iframe', async({page, server}) => {
+    server.setRoute('/x-frame-options-deny.html', async (req, res) => {
+      res.setHeader('Content-Type', 'text/html');
+      res.setHeader('X-Frame-Options', 'DENY');
+      res.end(`<!DOCTYPE html><html><head><title>login</title></head><body style="background-color: red;"><p>dangerous login page</p></body></html>`);
+    });
+    await page.goto(server.EMPTY_PAGE);
+    const refusalText = new Promise(f => {
+      page.on('console', msg => {
+        if (msg.text().match(/Refused to display/i))
```
Comment (rwoll) [inline test/frame.jest.js:277]: Chrome sometime logs an error about the favicon missing, so we filter the error messages.

File: test/frame.jest.js [hunk 2]
```diff
@@ -265,4 +265,20 @@ describe('Frame Management', function() {
...
+          f(msg.text());
+      });
+    });
+    await page.setContent(`<iframe src="${server.CROSS_PROCESS_PREFIX}/x-frame-options-deny.html"></iframe>`);
+    expect(await refusalText).toMatch(/Refused to display 'http.*\/x-frame-options-deny\.html' in a frame because it set 'X-Frame-Options' to 'deny'./i)
```
Comment (dgozman) [inline test/frame.jest.js:282]: Can we also check that `page.frames()[1].content()` does not contain 'dangerous login page'?

**Review by dgozman — APPROVED** [review]
Wonderful! Thank you for the PR.

File: test/frame.jest.js (hunk 2 above)
Comment (rwoll) [reply test/frame.jest.js:282]: If I add:

```
expect(await page.frames()[1].content()).not.toContain('dangerous login page');
```

with Chromium Headfull we crash with:

```
TypeError: Cannot read property 'content' of undefined
```

and with Webkit Headfull we timeout waiting for content.

Comment (dgozman) [reply test/frame.jest.js:282]: I see! That's unfortunate, but oh well.

Comment (rwoll) [reply test/frame.jest.js:282]: Yeah, the frame story is interesting (via `utils.dumpFrames(page.mainFrame())`):

Chromium:
```
[ 'http://localhost:<PORT>/empty.html' ]
```

Webkit:
```
[ 'http://localhost:<PORT>/empty.html', '    ' ]
```

Firefox:
```
    [
      'http://localhost:<PORT>/empty.html',
      '    http://127.0.0.1:<PORT>/x-frame-options-deny.html'
    ]
```

**Pull request merged**